tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/worker/WorkerTest.java

```diff
@@ -347,4 +348,16 @@ public void waitInReturn() {
         Assert.assertEquals(mapResult.get("w1").stringValue(), "w1");
         Assert.assertEquals(mapResult.get("w2").stringValue(), "w2");
     }
+
+    @Test(expectedExceptions = BLangRuntimeException.class)
+    public void testFunctionWithWorkerInsideLock() {
+        BValue[] returns = BRunUtil.invoke(result, "testPanicWorkerInsideLock");
+    }
+
+    @Test
+    public void testWorkerInsideLock() {
+        CompileResult result = BCompileUtil.compile("test-src/workers/worker-in-lock.bal");
+        int index = 0;
+        BAssertUtil.validateError(result, index++, 4, 20);
```

Comment:
Shall we validate the error message also?

Reply:
+1 will do